python_examples/rebound.py

```diff
@@ -127,4 +149,158 @@ def step():
+    _y  = star.y + r*(numpy.sin(Omega)*numpy.cos(omega+f) + numpy.cos(Omega)*numpy.sin(omega+f)*numpy.cos(inc))
+    _z  = star.z + r*numpy.sin(omega+f)*numpy.sin(inc)
+    
+    n = numpy.sqrt(get_G()*(star.m+mass)/(a**3))
+    
+    # Murray & Dermott Eq. 2.36 after applying the 3 rotation matrices from Sec. 2.8 to the velocities in the orbital plane
+    _vx = star.vx + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(-numpy.cos(inc)*numpy.cos(omega)*numpy.sin(Omega) - numpy.cos(Omega)*numpy.sin(omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.cos(Omega) - numpy.cos(inc)*numpy.sin(omega)*numpy.sin(Omega)))
+    _vy = star.vy + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(numpy.cos(inc)*numpy.cos(omega)*numpy.cos(Omega) - numpy.sin(omega)*numpy.sin(Omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.sin(Omega) + numpy.cos(inc)*numpy.cos(Omega)*numpy.sin(omega)))
+    _vz = star.vz + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*numpy.cos(omega)*numpy.sin(inc) - numpy.sin(f)*numpy.sin(inc)*numpy.sin(omega))
+    
+    return Particle(m  = mass, x = _x, y = _y, z = _z, vx = _vx, vy = _vy, vz = _vz)
+
+def p2orbit(p, star):
+    '''Takes 2 Particle objects (one for object and one for central body,
+        which hold Cartesian elements) and returns a Orbit object
+        with all the orbital elements'''
+    
+    if star.m <= 1.e-30: raise ValueError('Star has no mass.')
+    o = Orbit()
+    mu = get_G()*(p.m+star.m)
+    p.x -= star.x
+    p.y -= star.y
+    p.z -= star.z
+    p.vx -= star.vx
+    p.vy -= star.vy
+    p.vz -= star.vz
+    h0 = (p.y*p.vz - p.z*p.vy)                  # angular momentum vector
+    h1 = (p.z*p.vx - p.x*p.vz)
+    h2 = (p.x*p.vy - p.y*p.vx)
+    o.h = numpy.sqrt ( h0*h0 + h1*h1 + h2*h2 )  # abs value of angular moment
+    if o.h <= 1.e-30: raise ValueError('Particle orbit is radial.')
+    v = numpy.sqrt ( p.vx*p.vx + p.vy*p.vy + p.vz*p.vz )
+    o.r = numpy.sqrt ( p.x*p.x + p.y*p.y + p.z*p.z )
+    if o.r <= 1.e-30: raise ValueError('Particle and star positions are the same.')
+    vr = (p.x*p.vx + p.y*p.vy + p.z*p.vz)/o.r
+    e0 = 1./mu*( (v*v-mu/o.r)*p.x - o.r*vr*p.vx )
+    e1 = 1./mu*( (v*v-mu/o.r)*p.y - o.r*vr*p.vy )
+    e2 = 1./mu*( (v*v-mu/o.r)*p.z - o.r*vr*p.vz )
+    o.e = numpy.sqrt( e0*e0 + e1*e1 + e2*e2 )   # eccentricity
+    o.a = -mu/( v*v - 2.*mu/o.r )               # semi major axis
+    o.P = 2.*numpy.pi*numpy.sqrt( o.a*o.a*o.a/mu )  # period
+    o.inc = numpy.arccos( h2/o.h )              # inclination (wrt xy-plane)
+    # if pi/2<i<pi it's retrograde
+    n0 = -h1                                    # node vector
+    n1 =  h0                                    # in xy plane => no z component
+    n = numpy.sqrt( n0*n0 + n1*n1 )
+    er = p.x*e0 + p.y*e1 + p.z*e2
+    if (n<=1.e-30 or o.inc<=1.e-30):            # we are in the xy plane
+        o.Omega=0.
+        if (o.e <= 1.e-10):                     # omega not defined for circular orbit
```

Comment:
Not sure about this one. But `1e-10` seems odd. Either something closer to `1e-16` or `1e-308`.